allennlp/models/vilbert_new.py

```diff
@@ -0,0 +1,276 @@
+import collections
+import logging
+from copy import deepcopy
+from typing import Dict, List, Optional
+
+from overrides import overrides
+import torch
+
+from allennlp.data import TextFieldTensors, Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules import TimeDistributed
+from allennlp.training.metrics import CategoricalAccuracy
+
+from allennlp.modules.transformer import (
+    Embeddings,
+    TransformerEmbeddings,
+    BiModalEncoder,
+    ActivationLayer,
+    TransformerModule,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ImageFeatureEmbeddings(Embeddings):
+    """Construct the embeddings from image, spatial location (omit now) and
+    token_type embeddings.
+    """
+
+    def __init__(self, feature_dim: int, hidden_dim: int, dropout: float = 0.0):
+
+        image_embeddings = torch.nn.Linear(feature_dim, hidden_dim)
+        image_location_embeddings = torch.nn.Linear(4, hidden_dim)
+        embeddings = [image_embeddings, image_location_embeddings]
+        super().__init__(embeddings, hidden_dim, dropout)
+
+
+@Model.register("nlvr2_vilbert_new")
+@Model.register("nlvr2_vilbert_from_huggingface_new", constructor="from_huggingface_model_name")
+class Nlvr2Vilbert(Model, TransformerModule):
```

Comment:
Why is this in a new file, registered as "new"?  I would have expected you to just modify the file that was there.